dt-fragment-loader.c

```diff
@@ -535,8 +535,7 @@ static int dfl_new_fragment_id(
             "l%d_c%d",
             &((*fid)->location),
             &((*fid)->compat));
-    if (ret < 2) {
-        (*fid)->parse_error = ret;
+    if (((*fid)->parse_error = ret < 2)) {
```

Comment:
I have an aversion to assignment in conditionals :c

Reply:
Reasonable in this case, am now assigning parse_error before the conditional.